File: 3a.py

def L_or_R(left, right):
    visited = set()
    S = []
    for l in left:
        visited.add(l)
        S.append(l)
    for r in right:
        if r not in visited:
            S.append(r)
    
    return S

def main():
    left = [1, 2, 3, 4, 5]
    right = [1, 2, 0, 4, 5]
    result = L_or_R(left, right)
    print(result)

if __name__ == "__main__":
    main()

File: 3a.cpp
/* Name: Ivy Loi
Date: 4/24/2026
Homework 3, Problem 3a
CPSC 335 section 10
*/

#include <iostream>
#include <list>
#include <unordered_set>

std::list<int> LOrR(const std::list<int>& L, const std::list<int>& R) {
    std::unordered_set<int> visited;
    std::list<int> S;

    for(int l : L) {
        if (visited.find(l) == visited.end()) {
            S.push_back(l);
            visited.insert(l);
        }
    }
    for(int r : R) {
        if (visited.find(r) == visited.end()) {
            S.push_back(r);
            visited.insert(r);
        }
    }
    return S;
}
int main() {
    std::list<int> L = {1, 2, 3, 4, 5};
    std::list<int> R = {4, 5, 6, 7, 8};
    std::list<int> S = LOrR(L, R);
    for (int i : S) {
        std::cout << i << " ";
    }
    std::cout << std::endl;
    return 0;
}